### PR TITLE
Adding the components section along with components view

### PR DESCRIPTION
**Pull request by LevinGuy**
Hi @mrin9 ,

I have added an option to show the components section and to view them.
I think it is at least a start - I think it can be even better but have to being somewhere :)

As I said, it is good to have the components searchable, examples, schemes (models) and etc.. - it will allow to easily understand the API and navigate to the models in case needed.

### CODE REVIEW

File: src/utils/spec-parser.js [hunk 1]
```diff
@@ -155,6 +158,77 @@ function groupByPaths(openApiSpec) {
 }
 */
 
+function getComponents(openApiSpec) {
+  const components = [];
+  for (const component in openApiSpec.components) {
```
Comment (mrin9) [inline src/utils/spec-parser.js]: must ensure `openApiSpec.components` is not `undefined`

File: src/rapidoc.js [hunk 2]
```diff
@@ -66,6 +67,7 @@ export default class RapiDoc extends LitElement {
       allowSpecFileLoad: { type: String, attribute: 'allow-spec-file-load' },
       allowSearch: { type: String, attribute: 'allow-search' },
       allowServerSelection: { type: String, attribute: 'allow-server-selection' },
+      showComponents: { type: String, attribute: 'show-components' },
```
[the remark this answers is not in the document]
Comment (LevinGuy) [reply src/rapidoc.js]: will do

File: src/rapidoc.js [hunk 3]
```diff
@@ -122,19 +124,19 @@ export default class RapiDoc extends LitElement {
...
+                fg1: isValidHexColor(this.textColor) ? this.textColor : '',
+                headerColor: isValidHexColor(this.headerColor) ? this.headerColor : '',
+                primaryColor: isValidHexColor(this.primaryColor) ? this.primaryColor : '',
+                navBgColor: isValidHexColor(this.navBgColor) ? this.navBgColor : '',
+                navTextColor: isValidHexColor(this.navTextColor) ? this.navTextColor : '',
+                navHoverBgColor: isValidHexColor(this.navHoverBgColor) ? this.navHoverBgColor : '',
+                navHoverTextColor: isValidHexColor(this.navHoverTextColor) ? this.navHoverTextColor : '',
+                navAccentColor: isValidHexColor(this.navAccentColor) ? this.navAccentColor : '',
+            };
+            return html`
```
[the remark this answers is not in the document]
Comment (LevinGuy) [reply src/rapidoc.js]: done

File: src/rapidoc.js [hunk 4]
```diff
@@ -235,6 +238,15 @@ export default class RapiDoc extends LitElement {
           padding: 16px 10px;
           font-weight:bold;
         }
+        .nav-bar-header {
+          border-top: 1px solid var(--nav-hover-bg-color);
+          font-size: var(--font-size-regular);
+          padding: 10px 7px;
+          font-weight:bold;
+          background-color: var(--nav-text-color);
+          color: var(--nav-bg-color);
```
[the remark this answers is not in the document]
Comment (LevinGuy) [reply src/rapidoc.js]: done (hopefully)

File: src/rapidoc.js [hunk 5]
```diff
@@ -490,6 +510,7 @@ export default class RapiDoc extends LitElement {
           `
         }
         ${html`<div class='nav-scroll'>
+          <div id='link-general' class='nav-bar-header' @click='${(e) => this.scrollToEl(e)}'>General</div>
```
[the remark this answers is not in the document]
Comment (LevinGuy) [reply src/rapidoc.js]: done, will be shown only if showInfo is true.

File: src/utils/spec-parser.js [hunk 6]
```diff
@@ -31,30 +31,30 @@ export default async function ProcessSpec(specUrl, sortTags = false, sortEndpoin
...
+                                                    }
+                                                    // convertedSpec = await converter.convertObj(resolvedRefSpec.resolved, convertOptions);
+                                                    resolveOptions = {
+                                                      resolveCirculars: false,
+                                                      location: specLocation, // location is important to specify to resolve relative external file references when using JsonRefs.resolveRefs() which takes an JSON object
+                                                    };
+                                                    resolvedRefSpec = await JsonRefs.resolveRefs(convertedSpec.openapi, resolveOptions);
+                                                    // jsonParsedSpec = convertedSpec.openapi;
+                                                    jsonParsedSpec = resolvedRefSpec.resolved;
+                                                    */
```
[the remark this answers is not in the document]
Comment (LevinGuy) [reply src/utils/spec-parser.js]: done

File: src/utils/spec-parser.js (hunk 1 above)
Comment (LevinGuy) [reply src/utils/spec-parser.js]: done

**Comment by LevinGuy** [comment]
Hi @mrin9 ,

I have handled all the comments you commented on.
Also, I have added a new small feature: showExtraInfo
The name is not that good, I admit, but I think it will be beneficial since now the user can have more detailed documentation and the "table of contents" for this - for now only for the info.description.

This feature is taking the headers (H1, H2 or #, ##) from the markdown and puts them on the left menu with links to the correlating headers in the documentation.

Thanks,
Guy

**Comment by mrin9** [comment]
Thanks for the PR !!!
`showExtraInfo` is a cool feature, ran it through many of our users and most of the tech-writers were pretty excited about it. The fact that they can control the nav-bar items just using the markdown is a great way to arrange topics.

However, as I mentioned before there will be style changes based on the feedback from our users and UX.

there are certain changes to the implementation that I would like to see, but the fact that many users like to get `showExtraInfo` feature in ASAP , I am gonna merge it now and will update it with the needed changes.

**Pull request merged**